plugins/woocommerce/tests/php/includes/wc-order-functions-test.php

```diff
@@ -56,4 +56,44 @@ public function test_wc_restock_refunded_items_stock_metadata() {
 		$this->assertEquals( 0, (int) $line_item->get_meta( '_reduced_stock', true ) );
 		$this->assertEquals( 2, (int) $line_item->get_meta( '_restock_refunded_items', true ) );
 	}
+
+	/**
+	 * Test update_total_sales_counts and check total_sales after order reflection.
+	 *
+	 * Tests the fix for issue #23796
+	 */
+	public function test_wc_update_total_sales_counts() {
+
+		$product = WC_Helper_Product::create_simple_product();
+
+		WC()->cart->add_to_cart( $product->get_id() );
+
+		$order_id = WC_Checkout::instance()->create_order(
+			array(
+				'billing_email'  => 'a@b.com',
+				'payment_method' => 'dummy',
+			)
+		);
+
+		$this->assertEquals( 0, $product->get_total_sales() );
+
+		$order = new WC_Order( $order_id );
+
+		$order->update_status( 'processing' );
+		$this->assertEquals( 1, $product->get_total_sales() );
+
+		$order->update_status( 'cancelled' );
+		$this->assertEquals( 0, $product->get_total_sales() );
+
+		$order->update_status( 'processing' );
+		$this->assertEquals( 1, $product->get_total_sales() );
+
+		$order->update_status( 'completed' );
+		$this->assertEquals( 1, $product->get_total_sales() );
+
+		if ( $order->delete( true ) ) {
+			$this->assertEquals( 0, $product->get_total_sales() );
+		}
```

Comment:
This is great, but can we add one or more assertions relating to refunds to show our design choices as they relate to that scenario?

Reply:
@barryhughes I think since we didn't touch the refund scenario, there is no need for the test for it? specially the product can be refunded without changing the order status, so I think no need for it now until we will decrease the sales count after refund?

Reply:
@barryhughes I added 2 assertions to make sure that the transitions between refunded and processing statuses will not affect the sales count